docs/registry.md

```diff
@@ -0,0 +1,341 @@
+# Registry
+
+The registry is a place where applications metadata are stored and versioned. It can be used by a cozy 
+
+We define an application registry as an API. This should allow us to defer the real implementation of the registry storage and allow different store implementations.
+
+## Objects
+
+Two type of objects are managed in the registry: applications and versions.
+
+### Application
+
+An application object contains the following fields:
+
+- `name`: the application name
+- `type`: the application type ("webapp" or "konnector")
+- `editor`: the application editor name
+- `category`: the application category
+- `repository`: object with type and url of package repository
+- `license`: the application license
+- `versions`: a list of all versions from the stable channel
+- `versionsDev`: a list of all versions from the dev channel
+
+### Version
+
+An application version object contains the following fields:
+
+- `name`: the application name
+- `description`: description from the package.json
+- `version`: the version string
+- `created_at`: date of the release creation
+- `channel`: the channel of the version (stable or dev)
+- `url`: url of the tarball containing the application at specified version
+- `size`: the size of the application package (uncompressed) in bytes
+- `sha256`: the sha256 checksum of the application content
+- `permissions`: the permissions map contained in the manifest
+- `locales`: the availables locales and the associated translated description
+
+The version string should be of the exact following form:
+
+  - stable channel: `vX.Y.Z` where `X`, `Y` and `Z` are positive or null integers.
+  - dev channel: `vX.Y.Z-checksum` where `X`, `Y` and `Z` are positive or null integers and `checksum` is a unique identifier of the dev release (typacally a shasum of the git commit)
```

Comment:
i think we should be compatible with [semver](http://semver.org/):

- no `v` in version string
- `X.Y.Z-beta.Z` for the beta channel
- `X.Y.Z-dev.checksum` for the dev channel